caluma/caluma_form/migrations/0035_historicalanswer_history_question_type.py

```diff
@@ -0,0 +1,41 @@
+# Generated by Django 2.2.12 on 2020-06-19 14:21
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import migrations, models
+
+
+def add_initial_historical_question_type(apps, schema_editor):
+    HistoricalAnswer = apps.get_model("caluma_form", "HistoricalAnswer")
+    HistoricalQuestion = apps.get_model("caluma_form", "HistoricalQuestion")
+
+    for answer in HistoricalAnswer.objects.all():
```

Comment:
This *might* be possible to optimize by doing a raw SQL `UPDATE` statement. It would need some ugly `JOIN` magic but should speed things up considerably.

Something like
```sql
update caluma_historical_answer ha
left join (
    select * from caluma_historical_question hq on (
        hq.slug = ha.question_id and history_date <= ha.history_date
    )
    order by history_date desc
    limit 1
) hq_last
set ha.history_question_type = hq_last.type
where hq_last.type is not null
```
followed up by a simplified `update` to fill in the remaining answers where no historical question was found:
```sql
update caluma_historical_answer ha
left join caluma_form_question q on (q.slug = ha.question_id)
set ha.history_question_type = q.type
where ha.history_question_type is null
```

Reply:
Huh that's clever, I was thinking of an update but wouldn't figure out the ordering / limit 1 part. I will give it a try!

Actually the fallback should theoretically not be necessary, but it might be that downstream projects did not setup the history properly (like we did...).

Thanks!

Reply:
Ok just measured the whole thing, brings it down to 30 seconds!

Reply:
Awesome :)